tests/nxdata_test.py

```
@@ -0,0 +1,137 @@
+from scippneutron.file_loading._hdf5_nexus import LoadFromHdf5
+from scippneutron.file_loading._json_nexus import LoadFromJson
+from scippneutron import nexus
+import pytest
+
+
+@pytest.fixture(params=[(open_nexus, LoadFromHdf5()), (open_json, LoadFromJson(''))])
+def nexus_group(request):
+    return request.param
+
+
+def test_without_coords(nexus_group: Tuple[Callable, LoadFromNexus]):
+    resource, loader = nexus_group
+    builder = NexusBuilder()
+    da = sc.DataArray(
+        sc.array(dims=['xx', 'yy'], unit='m', values=[[1.1, 2.2], [3.3, 4.4]]))
+    builder.add_data(Data(name='data1', data=da))
+    with resource(builder)() as f:
+        data = nexus.NXroot(f, loader)['entry/data1']
+        loaded = data[...]
+        assert sc.identical(loaded, da)
+
+
+def test_with_coords_matching_axis_names(nexus_group: Tuple[Callable, LoadFromNexus]):
+    resource, loader = nexus_group
+    builder = NexusBuilder()
+    da = sc.DataArray(
+        sc.array(dims=['xx', 'yy'], unit='m', values=[[1, 2, 3], [4, 5, 6]]))
+    da.coords['xx'] = da.data['yy', 0]
+    builder.add_data(Data(name='data1', data=da))
+    with resource(builder)() as f:
+        data = nexus.NXroot(f, loader)['entry/data1']
+        loaded = data[...]
+        assert sc.identical(loaded, da)
+
+
+def test_guessed_dim_for_coord_not_matching_axis_name(
+        nexus_group: Tuple[Callable, LoadFromNexus]):
+    resource, loader = nexus_group
+    builder = NexusBuilder()
+    da = sc.DataArray(
+        sc.array(dims=['xx', 'yy'], unit='m', values=[[1, 2, 3], [4, 5, 6]]))
+    da.coords['xx'] = da.data['yy', 0]
+    da.coords['xx2'] = da.data['yy', 1]
+    da.coords['yy'] = da.data['xx', 0]
+    builder.add_data(Data(name='data1', data=da))
+    with resource(builder)() as f:
+        data = nexus.NXroot(f, loader)['entry/data1']
+        loaded = data[...]
+        assert sc.identical(loaded, da)
+
+
+def test_guessed_dim_for_2d_coord_not_matching_axis_name(
```

What about data with `shape=[2,2]`? If I read it correctly, the code would currently load this data but does not check if the indices are correct. I think it should raise an exception in this case.

---

Changed, but I am not sure it is the right way. Wouldn't it make sense to assume that it is not transposed?